src/ci_providers/provider_circleci.ts

```diff
@@ -30,22 +32,20 @@ function _getSHA(inputs: UploaderInputs): string {
 
 function _getSlug(inputs: UploaderInputs): string {
   const { args, environment: envs } = inputs
-  let slug = args.slug || ''
+
+  const slug = setSlug(
+    args.slug,
+    envs.CIRCLE_PROJECT_USERNAME,
+    envs.CIRCLE_PROJECT_REPONAME,
+  )
   if (slug !== '') {
     return slug
   }
-  if (envs.CIRCLE_PROJECT_REPONAME !== '') {
-    slug = `${envs.CIRCLE_PROJECT_USERNAME}/${envs.CIRCLE_PROJECT_REPONAME}`
-  } else {
-    if (envs.CIRCLE_REPOSITORY_URL) {
-      slug = `${envs.CIRCLE_REPOSITORY_URL.split(':')[1].split('.git')[0]}`
-    } else {
-      throw new Error(
-        'Unable to detect slug from env. Please set manually with the -r flag',
-      )
-    }
+
+  if (isSetAndNotEmpty(envs.CIRCLE_REPOSITORY_URL)) {
+    return `${envs.CIRCLE_REPOSITORY_URL?.split(':')[1].split('.git')[0]}`
```

Comment:
#### Regarding `split(':')[1]…`
This will evaluate to `undefined` if split() returns an array with only one element.

#### Regarding `…[1].split`
Due to the aforementioned detail, a TypeError will occur here when evaluating `undefined.split`.